bin/common/utils.js

```diff
@@ -43,17 +43,12 @@ exports.loadNuxtConfig = (argv) => {
   options.mode =
     (argv.spa && 'spa') || (argv.universal && 'universal') || options.mode
 
-  return options
-}
-
-exports.getLatestHost = (argv) => {
-  const port =
-    argv.port || process.env.NUXT_PORT || process.env.PORT || process.env.npm_package_config_nuxt_port
-  const host =
-    argv.hostname ||
-    process.env.NUXT_HOST ||
-    process.env.HOST ||
-    process.env.npm_package_config_nuxt_host
+  // Server options
+  if (!options.server) {
```

Comment:
Is this one necessary ?

Reply:
If I was right, the raw user's `nuxt.config.js` is processed by `bin/common/utils.js` first then passed through `new Nuxt(options)` into `lib/common/options.js` to apply default values.

So the `bin/common/utils.js` haven't apply the default yet, the `option.servers` might be missing.

https://github.com/nuxt/nuxt.js/pull/3701/commits/fcc60e8bad3f87316b5e832cc4aa946400bed23f got build error.

Reply:
The CI: https://circleci.com/gh/nuxt/nuxt.js/4861

Reply:
I mean maybe this is enough here:

```js
  options.server = {
    host: argv.host,
    port: argv.port,
  }
```

Reply:
Than this will override user settings. The user setting will never be applied.

Let me do a quick test first.

Reply:
Yes it will override user setting.

as the `nuxt.config.js` set to 
```
server: {
  port: 8000
}
```
`bin/nuxt examples/custom-port-host --port=8080` will started on 8080 which is right

but `bin/nuxt examples/custom-port-host` will started on 3000 which is not expected.

the original approach behavior correct

Reply:
I think it won't, if not command line host:

```js
// option.server is:
options.server = {
  host: undefined,
  port: 3000, // example
}

// then into options.js, processed by defaultsDeep, it becomes

options.server = {
  host: process.env.NUXT_HOST ||
      process.env.HOST ||
      process.env.npm_package_config_nuxt_host,
  port: 3000, // example
}
```

Reply:
So you should keep `options.server.port = argv.port || options.server.port` in order to pass the user-set port, right?